plugin/pkg/policy/rqdata.go

```diff
@@ -0,0 +1,215 @@
+		},
+		"name": func(state request.Request) string {
+			return state.Name()
+		},
+		"class": func(state request.Request) string {
+			return state.Class()
+		},
+		"proto": func(state request.Request) string {
+			return state.Proto()
+		},
+		"size": func(state request.Request) string {
+			return strconv.Itoa(state.Len())
+		},
+		"remote": func(state request.Request) string {
+			return addrToRFC3986(state.IP())
+		},
+		"port": func(state request.Request) string {
+			return addrToRFC3986(state.Port())
+		},
+		"rcode": func(state request.Request) string {
+			rcode := ""
+			rr, ok := state.W.(*dnstest.Recorder)
+			if ok {
+				rcode = dns.RcodeToString[rr.Rcode]
+			}
+			if rcode == "" {
+				rcode = strconv.Itoa(rr.Rcode)
+			}
+			return rcode
+		},
+		"rsize": func(state request.Request) string {
+			rsize := ""
+			rr, ok := state.W.(*dnstest.Recorder)
+			if ok {
+				rsize = strconv.Itoa(rr.Len)
+			}
+			return rsize
+		},
+		"duration": func(state request.Request) string {
+			duration := ""
+			rr, ok := state.W.(*dnstest.Recorder)
+			if ok {
+				duration = strconv.FormatFloat(time.Since(rr.Start).Seconds(), 'f', -1, 64) + "s"
+			}
+			return duration
+		},
+		">rflags": func(state request.Request) string {
+			flags := ""
+			rr, ok := state.W.(*dnstest.Recorder)
+			if ok && rr.Msg != nil {
+				flags = flagsToString(rr.Msg.MsgHdr)
+			}
+			return flags
+		},
+		">id": func(state request.Request) string {
+			return strconv.Itoa(int(state.Req.Id))
+		},
+		">opcode": func(state request.Request) string {
+			return strconv.Itoa(int(state.Req.Opcode))
+		},
+		">do": func(state request.Request) string {
+			return boolToString(state.Do())
+		},
+		">bufsize": func(state request.Request) string {
+			return strconv.Itoa(state.Size())
+		},
+
+		"server_ip": func(state request.Request) string {
+			return addrToRFC3986(state.LocalIP())
+		},
+		"server_port": func(state request.Request) string {
+			return addrToRFC3986(state.LocalPort())
+		},
+		"response_ip": func(state request.Request) string {
+			ip := getRespIP(state.Req)
+			if ip != nil {
+				return addrToRFC3986(ip.String())
+			}
+			return ""
+		},
+	}
+	return &RequestExtractorMapping{replacements, emptyValue}
+}
+
+// GetValue extract the data that is mapped to this name and return the corresponding value as a string
+// if that value is empty then the defaultValue is returned
+// Second parameter is a boolean that inform if the name itself is supported in the mapping
+func (rd *RequestDataExtractor) GetValue(name string) (string, bool) {
+	f, ok := rd.requester.replacements[name]
+	if ok {
+		v := f(rd.state)
+		if v != "" {
+			return v, true
+		}
+		return rd.requester.emptyValue, true
+	}
+	return "", false
+}
+
+func boolToString(b bool) string {
+	if b {
+		return "true"
+	}
+	return "false"
+}
+
+// flagsToString checks all header flags and returns those
+// that are set as a string separated with commas
+func flagsToString(h dns.MsgHdr) string {
```

Comment:
why does this needs to become a string? For metadata?

Reply:
Not related to metadata.
I just keep the same definition and format that is currently in "replacer" (for the Log plugin)